toolchain/check/handle_binding_pattern.cpp

```diff
@@ -200,7 +202,21 @@ auto HandleParseNode(Context& context, Parse::BindingPatternId node_id)
 
 auto HandleParseNode(Context& context,
                      Parse::CompileTimeBindingPatternId node_id) -> bool {
-  return HandleAnyBindingPattern(context, node_id, /*is_generic=*/true);
+  bool is_generic = true;
+  if (context.decl_introducer_state_stack().innermost().kind ==
+      Lex::TokenKind::Let) {
+    auto scope_inst = context.insts().Get(context.scope_stack().PeekInstId());
+    if (!scope_inst.Is<SemIR::InterfaceDecl>() &&
+        !scope_inst.Is<SemIR::FunctionDecl>()) {
+      CARBON_DIAGNOSTIC(LetCompileTimeBindingUnsupported, Error,
+                        "`let` compile time bindings require function or "
+                        "interface (may be supported later)");
+      context.emitter().Emit(node_id, LetCompileTimeBindingUnsupported);
```

Comment:
Maybe this should be a `context.TODO` since the generics design does allow compile-time bindings in other scopes.
```suggestion
      context.TODO(node_id,
                   "`let` compile time binding outside function or interface");
```